Add ability to allow ssh commands only for specific users

cadamsdotcom opened the pull request:
This PR will allow only certain users (eg. CI, ansible) to run ssh commands on bastions, rather than blanket allowing all users.

Commit:
Add ability to allow ssh commands only for specific users

Commit:
Support instance refresh

Commit:
Trigger instance refresh on launch template version updates

tonycosentini committed:
Disable metadata endpoints on bastion instances.

tonycosentini committed:
Allow metadata endpoint with session tokens.

thecadams commented:
Hi, just wondering if anyone has had a chance to look at this PR in the past few months?

ethervoid committed:
Allow AWS provider v5

Signed-off-by: Mario de Frutos <mario@defrutos.org>

ethervoid committed:
Set ACL private explicitely due to changes in version 5

In Terraform AWS provider vestion 5 the acl option in the aws_s3_object
doesn't have default value set to private so we need to add it
explicitly

Signed-off-by: Mario de Frutos <mario@defrutos.org>

cadamsdotcom committed:
Merge branch 'master' into allow-ssh-commands-for-specific-users

Commit:
Fix README.md

cadamsdotcom commented on main.tf:
```diff
@@ -13,6 +13,7 @@ data "aws_kms_alias" "kms-ebs" {
 }
 
 resource "aws_s3_object" "bucket_public_keys_readme" {
+  acl        = "private"
```
Pretty sure this was missing and is needed, but not sure if it belongs in the PR.

cadamsdotcom commented:
Hi @Guimove,

Thanks for the note. Finally got around to resolving the conflicts, hope it's OK to review, let me know if anything else is needed.

Thanks! 🙏

cadamsdotcom commented:
Hi @Guimove ,
Hope you're well. Thanks for the module, it's been working really well for us.

Any chance you've had a look at this PR yet? Would be great to not have to maintain our fork!
👍 

Thanks again.
Chris.
